### ISSUE TITLE
Private CodeSandbox infinitely reloads in Safari

### ISSUE THREAD

**Issue by stephenkao** [codesandbox/codesandbox-client#6606]
# 🐛 bug report

## Preflight Checklist

- [x] I have read the [Contributing Guidelines][contributing] for this project.
- [x] I agree to follow the [Code of Conduct][code_of_conduct] that this project
      adheres to.
- [x] I have searched the issue tracker for an issue that matches the one I want
      to file, without success.

## Description of the problem

Sandboxes marked as 'Private' (as opposed to 'Public' or 'Unlisted') infinitely reload on Safari, which makes working in them impossible.

## How has this issue affected you? What are you trying to accomplish?

I've recently tried to have pairing sessions via Live Mode with others who use Safari as their default browser, and the only workaround is to use Chrome or Firefox.  To be fair, it's not a big deal for my coworkers, but my company also uses it to conduct interviews so it'd be nice to not have to ask candidates to change browsers.

### To Reproduce

- Create any sandbox
- Change the permissions from 'Public' to 'Private'
- Reload the sandbox

Here's a video of changing to 'Private' and reloading.  It stops immediately when I change it back to 'Public':
![2022-04-06 11 51 35](https://user-images.githubusercontent.com/2421042/162016849-9f3a2333-3319-48fb-87c5-e10975697cef.gif)

### Your Environment

| Software         | Name/Version |
| ---------------- | ------------ |
| Сodesandbox      | Latest  |
| Browser          | Safari, 15.2 (17612.3.6.1.6)  |
| Operating System | macOS 12.1  |

[code_of_conduct]: https://github.com/codesandbox/codesandbox-client/blob/master/CODE_OF_CONDUCT.md
[contributing]: https://github.com/codesandbox/codesandbox-client/blob/master/CONTRIBUTING.md

**Comment by JamesACS** [comment]
Hey @stephenkao - Thank you for raising this. I've been able to reproduce it on our end and have created a bug issue.

As soon as I have any updates I'll respond here

**Comment by amstiel** [comment]
It seems there are two types of error.

`[Error] Refused to apply a stylesheet because its hash, its nonce, or 'unsafe-inline' appears in neither the style-src directive nor the default-src directive of the Content Security Policy.
`


`[Error] Fetch API cannot load https://codesandbox.io/api/v1/sandboxes/foo/bar due to access control checks.`

**Comment by filipelima18** [comment]
Hi @amstiel! Can you share the URL of the project where you're experiencing these errors and any steps we should take to reproduce them?

**Comment by amstiel** [comment]
@filipelima18 Yeah, sure! Just created fresh React sandbox Draft: https://codesandbox.io/p/sandbox/test-sandbox-vcgch2

Or you can just create a fresh Draft yourself, I think it will do the same.

**Comment by gbatterbee** [comment]
Also happens for me with this one by the react team. 
Ios18.3
https://codesandbox.io/p/sandbox/dscqv9?file=%2Fsrc%2Factions.js
